src/compiler/html/inline-esm-import.ts

```diff
@@ -56,50 +74,85 @@ export const optimizeEsmImport = async (
   return false;
 };
 
+/**
+ * Update all relative module specifiers in some JS code to instead be nested
+ * inside of a supplied directory, transforming e.g. all imports of the form
+ * `'./foo.js'` to `'/build/foo.js'`.
+ *
+ * @param code the code to transform
+ * @param newDir the directory which should be prepended to all module
+ * specifiers in the code
+ * @returns a manifest containing transformed code and a list of transformed
+ * module specifiers
+ */
 export const updateImportPaths = (code: string, newDir: string) => {
-  const orgImportPaths: string[] = [];
-  const tsSourceFile = ts.createSourceFile('module.ts', code, ts.ScriptTarget.Latest);
-  ts.transform(tsSourceFile, [readImportPaths(orgImportPaths)]);
+  const orgModulePaths = readModulePaths(code);
 
-  orgImportPaths.forEach((orgImportPath) => {
-    const newPath = replacePathDir(orgImportPath, newDir);
+  for (const orgImportPath of orgModulePaths) {
+    const newPath = updateImportPathDir(orgImportPath, newDir);
     if (newPath) {
-      code = code.replace(`"${orgImportPath}"`, `"${newPath}"`);
-      code = code.replace(`'${orgImportPath}'`, `'${newPath}'`);
+      code = code.replace(new RegExp(`"${orgImportPath}"`, 'g'), `"${newPath}"`);
+      code = code.replace(new RegExp(`'${orgImportPath}'`, 'g'), `'${newPath}'`);
```

Comment:
Do we need to keep the same quotation style that the file originally had? Otherwise we could instantiate a single regex to match the import with either quotation style

Reply:
yeah I suppose we don't - I kept the same quotation-style-preserving logic that was already present, but I really just did so because that was the precedent and not for any real reason

Reply:
Yeah, I don't think we need it, but I'm not opposed to keeping it either. The only thing I can think if is that different quote styles can be used to allow _another_ quote style to be used in a string:
```ts
const foo = "We are 'the' champions";
```
which I don't _think_ would be allowed in module paths (otherwise that'd be _very_ confusing for users!)

Reply:
ok maybe out of an abundance of caution I'll just leave it as-is for now